spec/duration.html

```diff
@@ -52,5 +87,182 @@ <h1>Duration Objects</h1>
+      initialize subclass instances with the necessary internal slots.
+    </p>
+    <emu-clause id="sec-temporal-duration-constructor-alg">
+      <h1>
+        Temporal.Duration (_years_, _months_, _days_, _hours_, _minutes_,
+        _seconds_, _milliseconds_, _microseconds_, _nanoseconds_)
+      </h1>
+      <p>
+        When the Temporal.Duration function is called, the following steps are
+        taken:
+      </p>
+      <emu-alg>
+        1. If NewTarget is *undefined*, then
+          1. Throw a *TypeError* exception.
+        1. Let _h_ be ? ToInteger(_hours_).
+        1. Let _m_ be ? ToInteger(_minutes_).
+        1. Let _s_ be ? ToInteger(_seconds_).
+        1. Let _ms_ be ? ToInteger(_milliseconds_).
+        1. Let _mis_ be ? ToInteger(_microseconds_).
+        1. Let _ns_ be ? ToInteger(_nanoseconds_).
+        1. Let _bt_ be ? BalanceTime(_h_, _m_, _s_, _ms_, _mis_, _ns_).
+        1. Let _y_ be ? ToInteger(_years_).
+        1. Let _mo_ be ? ToInteger(_months_).
+        1. Let _d_ be ? ToInteger(_days_).
+        1. Increment _d_ by bt.[[Days]].
+        1. Return ? CreateDuration(_y_, _mo_, _d_, _h_, _m_, _s_, _ms_, _mis_, _ns_).
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-properties-of-the-temporal-duration-constructor">
+    <h1>Properties of the Temporal.Duration Constructor</h1>
+    <p>
+      The value of the [[Prototype]] internal slot of the Temporal.Duration
+      constructor is the intrinsic object %FunctionPrototype%.
+    </p>
+    <p>The Temporal.Duration constructor has the following properties:</p>
+    <emu-clause id="sec-temporal-duration-prototype">
+      <h1>Temporal.Duration.prototype</h1>
+      <p>
+        The initial value of Temporal.Duration.prototype is
+        %Temporal.Duration.prototype%.
+      </p>
+      <p>
+        This property has the attributes { [[Writable]]: *false*, [[Enumerable]]:
+        *false*, [[Configurable]]: *false* }.
+      </p>
+    </emu-clause>
+    <emu-clause id="sec-temporal-duration-prototype-fromstring">
+      <h1>Temporal.Duration.fromString ( _isoString_ )</h1>
+      <p>
+        The `fromString` method takes one argument _isoString_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _isoString_ be ? ToString(_isoString_).
+        1. Return ? DurationFromString(_isoString_).
+      </emu-alg>
+    </emu-clause>
+    <emu-clause id="sec-temporal-duration-prototype-from">
+        <h1>Temporal.Duration.from ( _arg_ )</h1>
+        <p>
+          The `from` method takes one argument _arg_.
+          The following steps are taken:
+        </p>
+        <emu-alg>
+          1. If Type(_arg_) is String, then
+            1. Return ? DurationFromString(_arg_).
+          1. If Type(_arg_) is BigInt, then
+            1. Return ? CreateDuration(0, 0, 0, 0, 0, 0, 0, 0, _arg_).
+          1. If Type(_arg_) is Number, then
+            1. Return ? CreateDuration(0, 0, 0, 0, 0, 0, +_arg_, 0, 0).
```

Comment:
I think we should hold off on those. Overloading bigint and number is an antipattern anyway, and making them do something different entirely seems like a bad idea too.

Reply:
Just following the polyfill here. Maybe @pipobscure can explain his reasoning?